src/main/drivers/system.c

```diff
@@ -60,19 +60,21 @@ static volatile int sysTickPending = 0;
 
 timeUs_t micros(void)
 {
-    static int32_t oldDWT = 0;
-    static timeUs_t curMicros = 0;
-
+    static struct {
+        int32_t dwt;
+        timeUs_t micros;
+    } state = {0,0};
+    timeUs_t ret;
+    ATOMIC_BLOCK_NB(configMAX_SYSCALL_INTERRUPT_PRIORITY) {
```

Comment:
This should allow gcc to keep some variables in registers ... but seems to work only when inlining and than it makes gcc inline more aggressively (but something is wrong, using ATOMIC_BLOCK + ATOMIC_BARRIER also increases file size,even when it should have no effect ... )